src/jit/lsra.cpp

```diff
@@ -3007,14 +3225,18 @@ LinearScan::buildRefPositionsForNode(GenTree *tree,
         {
             if (produce != 0)
             {
-                stack->Push(LocationInfo(currentLoc, interval, tree));
+                LocationInfoList list(listNodePool.GetNode(currentLoc, interval, tree));
+                bool added = operandToLocationInfoMap.AddOrUpdate(tree, list);
+                (assert(added), (void)added);
```

Comment:
> But which compiler generates a warning in this case? It seems that neither VC++ nor Clang do this.

The unused value warning is turned on for Clang in the CoreFX native components; I assumed that it was also enabled here. You're right, though--it is not. Removed.